samples/nrf9160/multicell_location/src/main.c

```diff
@@ -75,6 +80,7 @@ static void lte_handler(const struct lte_lc_evt *const evt)
 		LOG_INF("RRC mode: %s",
 			evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED ?
 			"Connected" : "Idle");
+		atomic_set(&rrc_connected, evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED ? 1 : 0);
```

Comment:
it would be nice if there was an option for `lte_lc_neighbor_cell_measurement` to wait until RRC was idle, so it could all be handled by the lte_lc library.

Reply:
That's something we could consider, I guess. The function could have a timeout parameter to specify how long the call can wait for RRC to be idle before the neighbor cell measurement is started. Anyway, I think we can discuss that and make a separate PR later if necessary.

Reply:
```
/* Wait until RRC connection has been released, otherwise modem will not be able to
 * measure neighbor cells unless currently configured by the network.
 */
```
I don't think I've ever seen this information before.  Is this documented anywhere?
Perhaps this information should be added to the header comments of `lte_lc_neighbor_cell_measurement` ?  
I never get neighbor cell measurements here in Portland and this could be the issue.

Reply:
@jayteemo I had a look at adding an option to `lte_lc_neighbor_cell_measurement()` to wait for RRC idle. It's not as straight forward as I had hoped. My initial idea was to return from the function immediately and just delay the `LTE_LC_EVT_NEIGHBOR_CELL_MEAS` from the library. That would mean that the `%NCELLMEAS` AT command would not necessarily be sent when the function is called, but it could happen later, when the RRC connection is released. The problem is that currently errors for the AT command are returned from the function and the event does not have any status information, just the serving cell and neighbor cell info.

To avoid more drastic changes to the API, the function would have to block until LTE is in idle. I'm just worried that this might not be as convenient and usable for the clients.